portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -2,40 +2,39 @@
 * Purpose: contains various JQuery functions that listen for events on main portfolio page
 */
 
-import { update_background, NUM_IMAGES } from './background_functions.js';
+import { updateBackground, NUM_IMAGES } from './background_functions.js';
+import { preloadImages, detectMobileDevice } from './helper_functions.js';
 
-let profile_image_paths = ["images/noogler.gif", "images/zoebakercircle.png"];
+let profileImagePaths = ["images/noogler.gif", "images/zoebakercircle.png"];
 
+let currentImageIndex = 1; // start on second image so as not to repeat landing image
 
-// Loads image paths into an arbitrary array, in order to preload images for improved performance.
-function preload_images(image_paths){
-  let image_arr = new Array(image_paths.length);
-  for (let i = 0; i < image_paths.length; i++) {
-    let temp_image = new Image(); 
-    temp_image.src = image_paths[i]; 
-    image_arr[i] = temp_image;
-  }
-  return image_arr;
-}
-
-let current_image_index = 1; // start on second image so as not to repeat landing image
 $(document).ready(function(){
+  let isMobile = detectMobileDevice();
+
+  let iconImages = preloadImages(profileImagePaths);
+ 
+  if (!isMobile){
+    // if change background button is clicked, background image for the body is changed.
+    $("#change-background").click(function(){
+      updateBackground(currentImageIndex);
+     currentImageIndex = (currentImageIndex + 1) % NUM_IMAGES;
+    });
 
-  let icon_images = preload_images(profile_image_paths);
-
-  // if change background button is clicked, background image for the body is changed.
-  $("#change-background").click(function(){
-    update_background(current_image_index);
-    current_image_index = (current_image_index + 1) % NUM_IMAGES;
-  });
-
-  // if mouse enters profile image attribute, change image to animated noogler.gif. Once mouse leaves,
-  // change back to normal profile icon
-   $(".profile").hover(function(){
+    // if mouse enters profile image attribute, change image to animated noogler.gif. Once mouse leaves,
+    // change back to normal profile icon
+    $(".profile").hover(function(){
         $(".profile").attr("src", "images/noogler.gif")
     },
     function(){
         $(".profile").attr("src", "images/zoebakercircle.png")
     });
-
+  } else { // if mobile, actively click to change (hover does not work as expected)
+    $(".profile").toggle(function(){
+      $(".profile").attr("src", "images/noogler.gif")
+    },
+    function(){
+      $(".profile").attr("src", "images/zoebakercircle.png")
```

Comment:
You use these strings in multiple places now. Consider storing them in a constant variable

Reply:
Good idea.